app/models/stacks_media_token.rb

```diff
@@ -0,0 +1,91 @@
+# this class is used for representing a time-limited grant of authorization to
+# view a streaming resource.  the token object has fields representing the resource
+# (id, file_name) and the request origin (user_ip_addr). additionally,
+# it has a timestamp indicating when it was created.
+#
+# the object can be serialized as a signed/encrypted string representation of
+# itself.  the class can also create an instance of itself from such a representation.
+#
+# the class can also take an arbitrary string and a list of expected values for the
+# token's fields.  if the string can be decrypted successfully (proving stacks
+# minted it), the recovered hash can be checked against the provided list of values
+# and the current time.  if the expected values match the decrypted values and the
+# token is younger than the max age, it's good, otherwise, it's not.
+#
+# thus, stacks can mint a token string and give it to a user, who can give it
+# to a service that stacks trusts, to allow that service to fulfil a request
+# to stream a resource.
+#
+# the order of operations might be something like the following example:
+# * a user makes a request to a stacks URL for a streaming resource.
+# * stacks creates a StacksMediaToken and serializes it to its encrypted string form.
+# * the encrypted string form of the token is included as part of a redirect to wowza.
+# * wowza receives the request for the resource stream, with the encrypted string.
+# * wowza calls back to stacks, and passes along: encrypted token string, IP from which
+#   it received the request, id, and file name.
+# * stacks uses StacksMediaToken#verify_encrypted_token_string to determine whether
+#   the encrypted string represents a still-valid token for the given resource.
+# * if it does, it returns 200 OK to wowza, if it doesn't then it returns 403 Forbidden.
+# * wowza serves the stream or not based on the response it gets from stacks.
+# * this checking in wowza can be implemented via a wowza request processing plugin.
+#
+# note that there's no check for whether a token's been used (there's no record of the token
+# minting in memory or DB, and so nothing to mark).
+class StacksMediaToken
+  MAX_TOKEN_AGE = 1.second
```

Comment:
Perhaps the MAX_TOKEN_AGE value should be in the Settings yml?  1 second for tests, but xxx recommended for in situ?